docs/chapter_data_structure/data_and_memory.md

```diff
@@ -84,13 +84,19 @@ comments: true
 === "JavaScript"
 
     ```js title=""
-
+    // JavaScript 的数组可以自由存储各种基本数据类型和对象
+    // JavaScript 的基本数据类型中只有数字, 没有浮点数, 这里使用0.1作为示例
+    const array = [0, 0.1, 'a', false]
     ```
 
 === "TypeScript"
 
     ```typescript title=""
-
+    // TypeScript 可以定义 JavaScript 数组的类型
+    const numbers: number = new Array(5).fill(0);
+    const decimals: number = new Array(5).fill(0.1);
+    const characters: string = new Array(5).fill('a');
+    const booleans: boolean = new Array(5).fill(true);
```

Comment:
```suggestion
    /* 使用多种「基本数据类型」来初始化「数组」 */
    const numbers: number[] = [];
    const strings: string[] = [];
    const booleans: boolean[] = [];
```

Reply:
It seems not right. In TypeScript, we can initiate different types of arrays.

Reply:
In my opinion, TypeScript can't initialize arrays, JavaScript can. TypeScript provides the data type for JavaScript.
![image](https://user-images.githubusercontent.com/64182179/212479436-5d0c785f-8b3f-4362-a570-0abea19a2c40.png)

Reply:
Because every TypeScript code will compile to JavaScript code, even if you give the TypeScript the wrong type like const a: wrong[] = [], it will still compile to const a = []. TypeScript type will not affect the compilation. But we need to know that TypeScript is used for us to check the type before runtime, so it will give us error messages on IDE.

Reply:
Yes, you are right. Because every TypeScript code will compile to JavaScript code, even if you give the TypeScript the wrong type like const a: wrong[] = [], it will still compile to const a = []. TypeScript type will not affect the compilation. 

However, we need to know that TypeScript is used for us to check the type before runtime, so it will give us error messages on IDE.  For example in the [TypeScript Playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBGCuBbARgUwE4QFxyW9A2gLowC8MxAUAihhAQAwnkCM1edjzMA5AEw9KQA):
<img width="614" alt="image" src="https://user-images.githubusercontent.com/24556310/212481607-6e04c93e-71f8-4891-9591-23359dcf8437.png">

<img width="528" alt="image" src="https://user-images.githubusercontent.com/24556310/212481573-6dd0293e-8649-4622-86cc-cd48cf2c00e0.png">

Reply:
I've fixed all.
But I think initialization isn't applicable to TS in `/* 使用多种「基本数据类型」来初始化「数组」 */`.
Do you think so?

Reply:
Yes, but we can do this. So I think it is ok here.

Reply:
OK.  There is nothing to modify here. You can merge it.
And I realize that both of us are in the same city. Maybe we can have have a cup of coffee in our spare time. @justin-tse

Reply:
Great! @zhuoqinyue   Hi! @krahets If you don't have other advice for this PR, you could merge it.